sys/arm64/include/hypervisor.h

```diff
@@ -43,7 +43,8 @@
 
 /* CPTR_EL2 - Architecture feature trap register */
 #define	CPTR_RES0	0x7fefc800
-#define	CPTR_RES1	0x000033ff
+#define	CPTR_RES1	0x000031ff
+#define	CPTR_TC		0x00000200	/* Trap Capabilities */
```

Comment:
Condition based on __has_feature(capabilities)? Certainly CPTR_TC should be guarded, and CPTR_RES1 is used by locore so I assume it needs the old value for non-Morello, and perhaps it also wants CPTR_TC for Morello too.

Reply:
Does locore do the right thing or not with the reduced CPTR_RES1?

Reply:
(if the answer is "I don't know", that's fine, but you should add a suitable XXXAT either here or there)

Reply:
We can use capabilities at EL1 and below without raising an exception when the `CPTR_TC` bit is clear.

Reply:
Which means what for my question?

Reply:
On the current version of the patch `CPTR_RES1` has the `CPTR_TC` field clear in a Morello kernel and set in a legacy kernel so should be correct.

Reply:
So `CPTR_RES1` is "trap if the guest tries to use any feature we don't know about", i.e. since a CHERI-aware kernel knows about capabilities there is no need to make them trap in the guest any more?

Reply:
It is the fields that are marked as `RES1`, i.e. unknown fields that you should set to `1`. In this case bit 9 was `RES1` in armv8, and is now `TC` in Morello. The `TC` field is set to `1` when you want to trap capabilities at EL2 so a CHERI-aware kernel that understands capabilities shouldn't trap. The guest is the same kernel configuring the `CPTR_EL2` register. It starts in EL2, then drops to EL1 after configuring the hypervisor registers.